### PR TITLE
Dockerfile: remove inheritance of crops/yocto image

### PR DESCRIPTION
**Pull request by sashko**
The Yocto Project recommends using Docker containers from crops for
building Yocto based projects.

With @ee336cd96e we have started using crops/yocto:ubuntu-16.04-base
as a base for our container, which in theory was supposed to make
maintenance of our Dockerfile eaiser.

On practice, Docker container from crops happened to have lots
unnecessary packages installed, e.g. Fluxbox and VNC server.

By removing inheritance of crops/yocto:ubuntu-16.04-base we make our
image more lightweight and gain more control over its content.

Signed-off-by: Oleksandr Kravchuk <oleksandr.kravchuk@pelagicore.com>

### CODE REVIEW

**Comment by pookimon** [comment]
Also we can limit number of RUN calls in Dockerfile, as every call is separate layer which means our image becomes larger. Using && between commands we can at least decrease this number.

**Comment by sashko** [comment]
> Also we can limit number of RUN calls in Dockerfile, as every call is separate layer which means our image becomes larger. Using && between commands we can at least decrease this number.

Aight, how big of performance improvement would this be? Could you link us to any relevant info?

**Comment by pookimon** [comment]
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/

File: Dockerfile [hunk 1]
```diff
@@ -36,25 +64,33 @@ RUN pip3 install unittest-xml-reporting
...
+ARG groupid=1000
+
 # Make sure the user/groupID matches the UID/GID given to Docker. This is so that mounted
 # dirs will get the correct permissions
 RUN usermod --uid $userid yoctouser
 RUN groupmod --gid $groupid yoctouser
 RUN echo 'yoctouser:yoctouser' | chpasswd
 RUN echo 'yoctouser ALL=(ALL) NOPASSWD:SETENV: ALL' > /etc/sudoers.d/yoctouser
 
+USER yoctouser
```
Comment (thenor) [inline Dockerfile]: This is done here on line 86 and on line 91. Does it need to be done twice?

File: Dockerfile [hunk 2]
```diff
@@ -1,33 +1,61 @@
...
+        texinfo \
+        tmux \
+        unzip \
+        wget \
+        xz-utils
+
+RUN apt-get clean
+
+# Remove all apt lists to avoid build caching
+RUN rm -rf /var/lib/apt/lists/*
```
Comment (martin-ejdestig) [inline Dockerfile]: Why did you move this? Cleanup? Separate commit if "yes".

File: Dockerfile [hunk 3]
```diff
@@ -36,25 +64,32 @@ RUN pip3 install unittest-xml-reporting
...
+RUN /usr/sbin/locale-gen en_US.UTF-8
 RUN localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
 RUN echo 'export LC_ALL="en_US.UTF-8"' >> /etc/profile
 ENV LANG en_US.utf8
 
+RUN useradd -U -m yoctouser
+
+# Enables us to overwrite the user and group ID for the yoctouser. See below
+ARG userid=1000
+ARG groupid=1000
```
Comment (martin-ejdestig) [inline Dockerfile]: Why did you move this? (I guess intent of original author was probably to have ARG at top. Do not know for sure nor what best practice. Regardless, unnecessary to move them if not needed. If it is a cleanup it should be a separate commit.)

**Comment by martin-ejdestig** [comment]
> > Also we can limit number of RUN calls in Dockerfile, as every call is separate layer which means our image becomes larger. Using && between commands we can at least decrease this number.
> 
> Aight, how big of performance improvement would this be? Could you link us to any relevant info?

Please do this in a separate commit if you do it in this pr.

**Comment by sashko** [comment]
> Why do you add:
> 
> * debianutils
> * libegl1-mesa
> * python3-git
> * python3-jinja2
> * python3-pexpect
> 
> in this commit? Perhaps I missed something else when comparing. Can not see them in the crops Dockerfile nor in the `build-install-dumb-init.sh` script.
> 
> If you added them because we need them for something and they are not in the crops image, it should be a separate commit.

It's dependencies for `Warrior`, but yes, I could add them with a separate commit.

File: Dockerfile (hunk 2 above)
Comment (sashko) [reply Dockerfile]: It just made more sense for me that we cleanup cache once we're done with using `apt-get`.

Comment (martin-ejdestig) [reply Dockerfile]: You are not done with apt-get at this point though. So not only is it wrong, it is hidden together with an unrelated change. Please make separate commits for things like this.

**Comment by sashko** [comment]
Just to be clear: I have created clones of our nightly jobs and ran them on the updated container to make sure it does not break anything.
